XLS-0085d-token-escrow/README.md

```diff
@@ -0,0 +1,243 @@
+| `CancelAfter`  | False       | Number | UInt32        | (Optional) The time, in seconds since the Ripple Epoch, when this escrow expires. This value is immutable; the funds can only be returned to the sender after this time. Required when creating an Escrow with IOU or MPT |
+
+**Process Overview:**
+
+- **Deduction from Source:**
+  - **IOU Tokens**: The escrowed amount is deducted from the source account's trustline balance with the issuer.
+  - **MPTs**: The escrowed amount is deducted from the source account's MPT balance.
+- **Escrow Object Creation:**
+  - An `Escrow` ledger object is created, holding the specified amount and associated conditions.
+  - **Fields Set**:
+    - `Amount` (IOU or MPT)
+    - `TransferRate` (stored at the time of creation for both IOUs and MPTs)
+    - `IssuerNode` (if the issuer is neither the source nor the destination)
+
+**Failure Conditions:**
+
+1. **Issuer Does Not Allow Token Escrow or Transfer:**
+   - **IOU Tokens**: If the issuer's account does not have the `lsfAllowTokenEscrow` flag set, the transaction fails with `tecNO_PERMISSION`.
+   - **MPTs**:
+     - If the `MPTokenIssuance` of the token being escrowed lacks the `lsfMPTCanEscrow` flag, the transaction fails with `tecNO_PERMISSION`.
+     - If the `MPTokenIssuance` of the token being escrowed lacks the `lsfMPTCanTransfer` flag, the transaction fails with `tecNO_PERMISSION`.
+
+2. **Source Account Not Authorized to Hold Token:**
+   - If the issuer requires authorization and the source is not authorized, the transaction fails with `tecNO_AUTH`.
+
+3. **Source Account's Token Holding Issues:**
+   - **IOU Tokens**: If the source lacks a trustline with the issuer, the transaction fails with `tecNO_LINE`.
+   - **MPTs**: If the source does not hold the MPT, the transaction fails with `tecOBJECT_NOT_FOUND`.
+
+4. **Source Account is Frozen or Token is Locked:**
+   - If the token is frozen (IOU) or locked (MPT) for the source, the transaction fails with `tecFROZEN`.
+
+5. **Insufficient Spendable Balance:**
+   - If the source account lacks sufficient spendable balance, the transaction fails with `tecUNFUNDED`.
+
+**State Changes:**
+
+- **Transfer from Source to Issuer Holding:**
+  - **IOU Tokens**: Amount deducted from the source's trustline balance.
+  - **MPTs**: Amount moved to a holding controlled by the issuer; the issuer's total outstanding amount remains unchanged.
+- **Escrow Object Creation:**
+  - The `Escrow` ledger object includes:
+    - `CancelAfter`: When the Escrow Expires (Required on IOU/MPT)
+    - `Amount`: Tokens held in escrow.
+    - `TransferRate`: `TransferRate` (IOUs) or `TransferFee` (MPTs) at creation.
+    - `IssuerNode`: Reference to the issuer’s ledger node if applicable.
```

Comment:
Is `IssuerNode` a pointer to the issuer's `AccountRoot` entry, or does it point to something else?

Reply:
It points to the page where the escrow keylet was stored in the issuer directory

Reply:
Ah ok, that was actually my next question -- it sounds like all Escrows involving an issuer have their keylets stored in the issuer's owner directory, is that correct? Going back to [this thread](https://github.com/XRPLF/XRPL-Standards/pull/272#discussion_r1979979181), I think this means you could compute the total amount of IOUs issued by an issuer by iterating over the issuer's owner directory, summing up all RippleState balances and Escrowed amounts. Is my understanding correct there?

Reply:
Yes, if the issuer is the sender or the receiver there is a keylet stored in the issuers directory. 

As for the issuer's `escrowed` amount I'm not certain of the exact implementation until I write it...

Reply:
@nkramer44 Can this be resolved?

Reply:
Bump @nkramer44

Reply:
Resolving for now; we can revisit this later if desired.